### PR TITLE
Add Multiline Text Field

### PR DESCRIPTION
**Pull request by mchernikov80**
image

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -636,4 +636,81 @@ entityOptionsBuilder.ConfigureProperty(shop => shop.Name, builder =>
 {
     builder.SetTruncationMaxCharacters(25);
 });
+```
+
+# Multiline Text Field property
+You can mark a property as multiline text field. Is allows inpur text in the several text rows. Disabled by default.
```
Comment (vasiliy-chefonov) [inline README.md:642]:
```suggestion
You can mark a property as multiline text field. It allows input of text in several text rows. Disabled by default.
```

File: README.md [hunk 2]
```diff
@@ -636,4 +636,81 @@ entityOptionsBuilder.ConfigureProperty(shop => shop.Name, builder =>
...
+public required string Street { get; set; }
+```
+
+## IsAutoGrow property identifies whether the height of the text field automatically changes with the number of lines of text.
+### Using Fluent API
+
+```csharp
+entityOptionsBuilder.ConfigureProperty(address => address.Street, builder =>
+{
+    builder.SetIsMultiline(true);
```
Comment (vasiliy-chefonov) [inline README.md:706]: What if we'll combine all these configurations in a single method?

Like: `optionsBuilder.SetIsMultiline(autoGrow: false, lines: 5, maxLines: 10);`?

File: README.md [hunk 3]
```diff
@@ -636,4 +636,81 @@ entityOptionsBuilder.ConfigureProperty(shop => shop.Name, builder =>
...
+    builder.SetMaxLines(15); // sets the max lines value as 15
+});
+```
+
+### Using Attribute
+
+```csharp
+[NetForgeProperty(IsMultiline = true, MaxLines = 15)]
```
Comment (vasiliy-chefonov) [inline README.md:696]: Let's create a separate attribute like `MultilineFieldAttribute`. with `lines` , `maxLines`, `isAutoGrow` properties.

File: src/Saritasa.NetForge.Domain/Attributes/NetForgePropertyAttribute.cs [hunk 4]
```diff
@@ -63,4 +63,16 @@ public class NetForgePropertyAttribute : Attribute
 
     /// <inheritdoc cref="PropertyMetadataBase.TruncationMaxCharacters"/>
     public int TruncationMaxCharacters { get; set; }
+
+    /// <inheritdoc cref="PropertyMetadataBase.IsMultiline"/>
+    public bool IsMultiline { get; set; }
+
+    /// <inheritdoc cref="PropertyMetadataBase.Lines"/>
+    public int Lines { get; set; } = 5;
```
Comment (vasiliy-chefonov) [inline src/Saritasa.NetForge.Domain/Attributes/NetForgePropertyAttribute.cs:71]: Do we need to set a default value here?

File: README.md [hunk 5]
```diff
@@ -636,4 +636,81 @@ entityOptionsBuilder.ConfigureProperty(shop => shop.Name, builder =>
...
+```
+
+### Using Attribute
+
+```csharp
+[NetForgeProperty(IsMultiline = true)]
+public required string Street { get; set; }
+```
+
+Also, there are some auxalary additional properties that are used in the multiple text field
```
Comment (AlexanderMakarov124) [inline README.md:662]: auxiliary?
```suggestion
Also, there are some auxiliary properties that are used in the multiple text field.
```

File: src/Saritasa.NetForge.DomainServices/PropertyOptionsBuilder.cs [hunk 6]
```diff
@@ -188,4 +188,40 @@ public PropertyOptionsBuilder SetTruncationMaxCharacters(int truncationMaxCharac
...
+    /// Marks this property as multiline.
+    /// </summary>
+    public PropertyOptionsBuilder SetIsMultiline(bool isMultiline)
+    {
+        options.IsMultiline = isMultiline;
+        return this;
+    }
+
+    /// <summary>
+    /// Sets default amount of lines.
```
Comment (AlexanderMakarov124) [inline src/Saritasa.NetForge.DomainServices/PropertyOptionsBuilder.cs:202]: What means `default` here?

**Review by AlexanderMakarov124 — APPROVED** [review]

**Pull request merged**